qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
It is still ambiguous as it is not assured to be true.

My concern is partially for this method, but more generally for qiita. The more we infer state that is vital for data integrity from an unstructured string (and in this PR it is untrusted as well), the more we open the door to very bad things happening in the future.

In this case, the most ideal solution is to never allow a user to upload something that has IDs of the form "study_id.sample_id". 

Within the rest of qiita, we should only ever manage sample IDs internally as a structured datatype that reflects its aggregate nature. We should only combine these datatypes at the last possible moment, and only when necessary.

I disagree that being lax with core datatypes that are vital to the integrity of our data model is a solution particularly as we're developing a system that will need to maintain its integrity for years.